apps/web/app/(navigation)/CreateNewBudget.tsx

```diff
@@ -17,118 +15,350 @@ import {
   SeparatorStyled,
   ButtonWrapperStyled,
   SeparatorStyledTop,
+  IconPickerStyled,
+  TextareaErrorStyled,
+  TextAreaWrapperStyled,
+  DatePickerWrapperStyled,
+  DatePickerErrorStyled,
+  FooterStyled,
+  ContentStyled,
 } from "./CreateNewBudget.styled";
 import { Form, Field } from "houseform";
-import { z } from "zod";
+import { useTranslate } from "lib/hooks";
+import { useValidateBudgetModal } from "./useValidateBudgetModal";
+import * as Tabs from "@radix-ui/react-tabs";
+import { LanguageContext } from "lib/contexts";
 
 type NewBudget = {
   onClose?: Function;
 };
 
+type currencyType = {
+  tag: string;
+  locale?: string;
+};
+
+type newBudgetType = {
+  name: string;
+  limit: number | string;
+  description: string;
+  icon: string;
+  dateStart: number | null;
+  dateEnd: number | null;
+  currency: currencyType;
+};
+
+const icons: IconType[] = [
+  "savings",
+  "directions_car",
+  "payments",
+  "subscriptions",
+  "shopping_cart",
+  "home",
+  "wallet",
+  "error",
+  "help",
+];
+
+//mocked existing user budgets
+export const loggedUserExistingBudgets = ["smutnarzaba", "frytki123"];
+//mocked currency
+export const acceptedCurrencies: Array<string> = ["USD", "PLN", "EUR", "GBP"];
+
 export const CreateNewBudget = ({ onClose }: NewBudget) => {
   const [defaultValue, setDefaultValue] = useState("settings");
-  const [selectedIcon, setSelectedIcon] = useState<Icon | undefined>(
-    "settings"
-  );
+  const [selectedIcon, setSelectedIcon] = useState<IconType>("savings");
+  const [lang, setLang] = useState<string>("en-US");
 
-  const icons: IconType[] = [
-    "savings",
-    "directions_car",
-    "payments",
-    "subscriptions",
-    "shopping_cart",
-    "home",
-    "wallet",
-    "error",
-    "help",
-  ];
+  const { t, dict } = useTranslate("AddNewBudgetModal");
+  const { currentLang } = useContext(LanguageContext);
 
-  const budgetNames = ["smutnarzaba", "frytki123"];
+  const {
+    checkNameOnChange,
+    checkNameOnSubmit,
+    checkCurrency,
+    checkLimit,
+    checkDescription,
+    checkStartDate,
+    checkEndDate,
+  } = useValidateBudgetModal("AddNewBudgetModal");
+
+  const [newBudget, setNewBudget] = useState<newBudgetType>({
+    name: "",
+    limit: "",
+    description: "",
+    icon: selectedIcon,
+    dateStart: null,
+    dateEnd: null,
+    currency: {
+      tag: "",
+      locale: lang,
+    },
+  });
+
+  const onSelectStartDate = (date: Date) => {
```

Comment:
changes to fix the clear icon in date picker inputs (5/6)

```suggestion
  const onSelectStartDate = (date: Date | null) => {
```